sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentCollection.java

```diff
@@ -411,6 +413,36 @@ public void setClientEncryptionPolicy(ClientEncryptionPolicy value) {
         setProperty(this, Constants.Properties.CLIENT_ENCRYPTION_POLICY, value);
     }
 
+    /**
+     * Gets the Vector Embedding Policy containing paths for embeddings along with path-specific settings for the item
+     * used in performing vector search on the items in a collection in the Azure CosmosDB database service.
+     *
+     * @return the Vector Embedding Policy.
+     */
+    public VectorEmbeddingPolicy getVectorEmbeddingPolicy() {
+        if (this.vectorEmbeddingPolicy == null) {
+            if (super.has(Constants.Properties.VECTOR_EMBEDDING_POLICY)) {
+                this.vectorEmbeddingPolicy = super.getObject(Constants.Properties.VECTOR_EMBEDDING_POLICY,
+                    VectorEmbeddingPolicy.class);
+            }
+        }
+        return this.vectorEmbeddingPolicy;
+    }
+
+    /**
+     * Sets the Vector Embedding Policy containing paths for embeddings along with path-specific settings for the item
+     * used in performing vector search on the items in a collection in the Azure CosmosDB database service.
+     *
+     * @param value the Vector Embedding Policy.
+     */
+    public void setVectorEmbeddingPolicy(VectorEmbeddingPolicy value) {
+        if (value == null) {
+            throw new IllegalArgumentException("VectorEmbeddingPolicy cannot be null.");
```

Comment:
Throw NullPointerException instead.

Reply:
Please use checkNotNull helper function - that will ensure consistent type of exceptions across